janus_core/helpers/utils.py

```diff
@@ -432,3 +437,83 @@ def check_files_exist(config: dict, req_file_keys: Sequence[PathLike]) -> None:
         # Only check if file key is in the configuration file
         if not Path(config[file_key]).exists():
             raise FileNotFoundError(f"{config[file_key]} does not exist")
+
+
+def validate_slicelike(maybe_slicelike: SliceLike) -> None:
+    """
+    Raise an exception if slc is not a valid SliceLike.
+
+    Parameters
+    ----------
+    maybe_slicelike : SliceLike
+        Candidate to test.
+
+    Raises
+    ------
+    ValueError
+        If maybe_slicelike is not SliceLike.
+    """
+    if isinstance(maybe_slicelike, (slice, range, int)):
+        return
+    if isinstance(maybe_slicelike, tuple) and len(maybe_slicelike) == 3:
```

Comment:
What if it's a `list` or some other sequence like I've done a `map`? Would it be better to use `itertools.islice` to take the front 3, check the iterator's empty and then check the properties?

**N.B.** being restrictive is fine, it's just whether we want to be.

Reply:
At the moment I have been following the definition hence the restriction. ```list``` could be included, but then will this effect post-processing and any other users of ```SliceLike```? Perhaps those functions do work with ```list``` but I have not tried. I don't have strong opinions either way.

```
StartStopStep = tuple[Optional[int], Optional[int], int]
SliceLike = Union[slice, range, int, StartStopStep]
```

Reply:
We can always expand them later. It's not too hard.